extensions/wikia/UserLogin/tests/UserLoginTest.php

```diff
@@ -313,114 +315,110 @@ public function testMailPassword( $requestParams, $mockWgUserParams, $mockAuthPa
+		$mockAuthParams2 = [ 'allowPasswordChange' => false ];
 		$expMsg2 = wfMessage( 'userlogin-error-resetpass_forbidden' )->escaped();
 
 		// user is blocked
-		$mockWgUserParams3 = array( 'isBlocked' => true );
-		$mockAuthParams3 = array( 'allowPasswordChange' => true );
+		$mockWgUserParams3 = [ 'isBlocked' => true ];
+		$mockAuthParams3 = [ 'allowPasswordChange' => true ];
 		$expMsg3 = wfMessage( 'userlogin-error-blocked-mailpassword' )->escaped();
 
 		// user not found
-		$mockWgUserParams4 = array(
+		$mockWgUserParams4 = [
 			'isBlocked' => false,
-		);
+		];
 		$mockUserParams4 = false;
 		$expMsg4 = wfMessage( 'userlogin-error-noname' )->escaped();
 
 		// User - invalid user (user id = 0)
-		$mockUserParams5 = array(
+		$mockUserParams5 = [
 			'load' => null,
 			'loadFromDatabase' => null,
 			'getId' => 0
-		);
+		];
 		$expMsg5 = wfMessage( 'userlogin-error-nosuchuser', $reqParams2['username'] )->escaped();
 
 		// User - password reminder throttled
-		$mockUserParams6 = array(
+		$mockUserParams6 = [
 			'load' => null,
 			'loadFromDatabase' => null,
 			'getId' => $testUserId,
 			'isPasswordReminderThrottled' => true
-		);
+		];
 		$expMsg6 = wfMessage( 'userlogin-error-throttled-mailpassword', round( F::app()->wg->PasswordReminderResendTime, 3 ) )->escaped();
 
 		// User - mail error
-		$mockUserParams7 = array(
+		$mockUserParams7 = [
 			'load' => null,
 			'loadFromDatabase' => null,
 			'getId' => $testUserId,
 			'isPasswordReminderThrottled' => false,
-			'mockValueMap' => array(
-				'getOption' => array(
-					array( UserLoginSpecialController::NOT_CONFIRMED_SIGNUP_OPTION_NAME, null, false, true ),
-					array( 'language', null, false, 'en' )
-				)
-			)
-		);
-		$status7 = Status::newFatal( '' );
-		$mockLoginFormParams7 = array( 'mailPasswordInternal' => $status7 );
-		$expMsg7 = wfMsgExt( 'userlogin-error-mail-error', array( 'parseinline' ), $status7->getMessage() );
+			'params' => [
+				'mName' => 'WikiaUser',
+				'mEmail' => self::TEST_EMAIL,
+			],
+			'mockValueMap' => [
+				'getOption' => [
+					[ UserLoginSpecialController::NOT_CONFIRMED_SIGNUP_OPTION_NAME, null, false, true ],
+					[ 'language', null, false, 'en' ]
+				]
+			]
+		];
 
 		// User - email sent
-		$status8 = Status::newGood();
-		$mockLoginFormParams8 = array( 'mailPasswordInternal' => $status8 );
 		$expMsg8 = wfMessage( 'userlogin-password-email-sent', $reqParams2['username'] )->escaped();
 
 		// User - mail error
-		$mockUserParams9 = array(
+		$mockUserParams9 = [
 			'load' => null,
 			'loadFromDatabase' => null,
 			'getId' => $testUserId,
 			'isPasswordReminderThrottled' => false,
-			'params' => array(
-				'mId' => $testUserId,
-				'mName' => 'WikiaUser'
-			),
-			'mockValueMap' => array(
-				'getOption' => array(
-					array( UserLoginSpecialController::NOT_CONFIRMED_SIGNUP_OPTION_NAME, null, false, true ),
-					array( 'language', null, false, 'en' )
-				)
-			)
-		);
+			'params' => [
```

Comment:
The DB for user ```WikiaUser``` doesn't have this ID anymore and I don't know why both have to be specified.  Removed it and added mEmail which the new controller checks for.